### PR TITLE
Integrate new observer system

### PR DESCRIPTION
**Pull request by basine**
Needs release with https://github.com/wireapp/wire-ios-data-model/pull/138

### CODE REVIEW

File: Source/Transcoders/ImageDownloadRequestStrategy.swift [hunk 1]
```diff
@@ -111,12 +111,10 @@ extension ImageDownloadRequestStrategy : ZMDownstreamTranscoder {
...
         
-        let uiMOC = managedObjectContext.zm_userInterface
-        
-        uiMOC?.performGroupedBlock { 
-            guard let message = try? uiMOC?.existingObject(with: message.objectID) else { return }
-            uiMOC?.globalManagedObjectContextObserver.notifyNonCoreDataChangeInManagedObject(message!)
-        }
+        guard let uiMOC = managedObjectContext.zm_userInterface else { return }
+        NotificationDispatcher.notifyNonCoreDataChanges(objectID: message.objectID,
+                                                        changedKeys: ["mediumData", ZMAssetClientMessageDownloadedImageKey],
```
Comment (marcoconti83) [inline Source/Transcoders/ImageDownloadRequestStrategy.swift]: Do we have a constant to replace the magic string?

File: Source/Transcoders/LinkPreviewAssetDownloadRequestStrategy.swift [hunk 2]
```diff
@@ -93,12 +93,10 @@ import WireRequestStrategy
...
         guard success! else { return }
         
-        let uiMOC = managedObjectContext.zm_userInterface
-        let objectID = message.objectID
-        uiMOC?.performGroupedBlock {
-            guard let uiMessage = try? uiMOC?.existingObject(with: objectID) else { return }
-            uiMOC?.globalManagedObjectContextObserver.notifyNonCoreDataChangeInManagedObject(uiMessage!)
-        }
+        guard let uiMOC = managedObjectContext.zm_userInterface else { return }
+        NotificationDispatcher.notifyNonCoreDataChanges(objectID: message.objectID,
```
Comment (marcoconti83) [inline Source/Transcoders/LinkPreviewAssetDownloadRequestStrategy.swift]: Same as above

File: Source/Transcoders/ImageDownloadRequestStrategy.swift (hunk 1 above)
Comment (basine) [inline Source/Transcoders/ImageDownloadRequestStrategy.swift]: We might...

Comment (basine) [inline Source/Transcoders/ImageDownloadRequestStrategy.swift]: No, we don't

Comment (basine) [inline Source/Transcoders/ImageDownloadRequestStrategy.swift]: Unless you want to re-release ZMCDataModel

Comment (daehn) [inline Source/Transcoders/ImageDownloadRequestStrategy.swift]: `#keyPath` could work if mediumData is publicly exposed.

**Pull request merged**